src/MB_Toolbox.php

```diff
@@ -382,4 +383,49 @@ private function authenticate() {
     $this->auth = $auth;
   }
 
+   /**
+   * Generate user specific link URL to user subscription setting
+   * (http://subscriptions.dosomething.org) web page. The page is an interface
+   * to allow users to subscribe/unsubscribe to different types of email
+   * messaging.
+   *
+   * @param string $targetEmail
+   *   The email address to generate the subscription URL for.
+   *
+   * @return string $subscription_link
+   *   The URL to the user subscription settings web page. The link includes a
+   *   key md5 hash value to limit page access to authorized users who have
+   *   received an email from the lists the subscription page administers.
+   */
+  public function subscriptionsLinkGenerator($targetEmail) {
+
+    $this->statHat->clearAddedStatNames();
+
+    $curlUrl = $this->settings['ds_drupal_api_host'];
+    $port = $this->settings['ds_drupal_api_port'];
+    if ($port != 0 && is_numeric($port)) {
```

Comment:
I'd change the first statement to `$port > 0` so the negative numbers will not pass also.